Source/Parallelization/WarpXComm.cpp

```diff
@@ -370,8 +370,12 @@ WarpX::FillBoundaryE (int lev, PatchType patch_type)
         }
 
         const auto& period = Geom(lev).periodicity();
-        Vector<MultiFab*> mf{Efield_fp[lev][0].get(),Efield_fp[lev][1].get(),Efield_fp[lev][2].get()};
-        amrex::FillBoundary(mf, period);
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+            ng <= Efield_fp[lev][0]->nGrowVect(),
+            "Error: in FillBoundaryE, requested more guard cells than allocated");
```

Comment:
Note: as an optimization in the future: could we try to overlap the asynchronous MPI Sends and MPI Receives for the 3 components of E and B here? Is there an `amrex` API for this?

Reply:
Good question, let's keep it in mind.